src/python/grpcio/grpc/beta/implementations.py

```diff
@@ -75,10 +75,73 @@ def ssl_client_credentials(root_certificates, private_key, certificate_chain):
   Returns:
     A ClientCredentials for use with an SSL-enabled Channel.
   """
-  intermediary_low_credentials = _intermediary_low.ClientCredentials(
-      root_certificates, private_key, certificate_chain)
+  return ClientCredentials(_low.channel_credentials_ssl(
+      root_certificates, private_key, certificate_chain))
+
+
+class CallCredentials(object):
+  """A value encapsulating data asserting an identity over an *established*
+  channel. May be composed with ClientCredentials to always assert identity for
+  every call over that channel.
+
+  This class and its instances have no supported interface - it exists to define
+  the type of its instances and its instances exist to be passed to other
+  functions.
+  """
+
+  def __init__(self, low_credentials):
+    self._low_credentials = low_credentials
+
+
+def metadata_call_credentials(metadata_callback, name=None):
+  """Construct call credentials from an auth metadata plugin.
+
+  Args:
+    metadata_callback (callable): Callback accepting an
+      interfaces.GRPCAuthMetadataContext and another callback (itself accepting a
+      list of 2-tuples of metadata key/value pairs and an Exception; the
+      Exception may be None).
+
+  Returns:
+    A CallCredentials object for use in a GRPCCallOptions object.
+  """
+  if name is None:
+    name = metadata_callback.__name__
+  return CallCredentials(
+      _low.call_credentials_metadata_plugin(metadata_callback, name))
+
+def composite_call_credentials(call_credentials, additional_call_credentials):
+  """Compose two call credentials to make a new one.
+
+  Args:
+    call_credentials: A CallCredentials object.
+    additional_call_credentials: Another CallCredentials object to compose on
+      top of call_credentials.
+
+  Returns:
+    A CallCredentials object for use in a GRPCCallOptions object.
+  """
+  return CallCredentials(
+      _low.call_credentials_composite(
+          call_credentials._low_credentials,
+          additional_call_credentials._low_credentials))
+
+def composite_client_credentials(client_credentials,
+                                 additional_call_credentials):
+  """Compose call credentials on top of client credentials to make a new one.
+
+  Args:
+    client_credentials: A ClientCredentials object.
+    additional_call_credentials: A CallCredentials object to compose on
+      top of client_credentials.
+
+  Returns:
+    A ClientCredentials object for use in a GRPCCallOptions object.
+  """
   return ClientCredentials(
-      intermediary_low_credentials._internal, intermediary_low_credentials)  # pylint: disable=protected-access
+      _low.call_credentials_composite(
```

Comment:
are you sure it's `_low.call_credentials_composite` here and not `_low.client_credentials_composite` (which should be changed to `_low.channel_credentials_composite` if that is the case).